swfo/brdf_fallback.py

```diff
@@ -400,34 +409,25 @@ def __get_data(dat_filename, param, window):
     return np.nanstd(data, axis=0, ddof=1, keepdims=False)
 
 
-def concatenate_files(infile_paths, outfile, h5_info, doy):
+def concatenate_files(infile_paths, outfile, h5_info, doy, tile_metadata):
     """
     A function to concatenate multiple h5 files
     """
     assert len(infile_paths) == 7
-
-    # TODO create dataset to store uuid for brdf_fallback provenance (use h5_info 
-    # for to id files used in  threshold generation)
-    # h5_info is a dict with key, val pairs: eg : key = '2002.01.01', 
-    # value = 'absolute path to a h5 file'. 
-
-    # if we want to store medata data for averages produced for each day of year
-    # then use following average_metadata dict, which contains the absolute path 
-    # of the files used in average brdf generation. 
-
-    # average_metadata = {key: h5_info[key] for key in h5_info if folder_doy(key) == doy}
-
-    # TODO write dataset called METADATA in outfile (final h5 files containting the 
-    # average brdf datasets for all seven MODIS bands. 
-
+    
+    # set the uuids used in processing average for given day of year
+    tile_metadata['lineage']['doy_average'] = [munge_metadata(h5_info[key]) 
+                                               for key in h5_info if folder_doy(key) == doy]
     with h5py.File(outfile, 'w') as out_fid:
+        write_h5_md(out_fid, [tile_metadata], ['/'])
```

Comment:
I'd write the metadata after writing the bands/array datasets.

Reasoning is that in a future state of the h5py library I expect us to be able to control/dictate the ordering (through creation order) for listing/traversing a dataset in a h5 archive (there is an existing bug/issue in h5py in that it's not being correctly tracked for the root of the archive; and it's only a partial implementation (not across all methods).